form.yml

```diff
@@ -0,0 +1,326 @@
+netweaver:
+  $name: Netweaver
+  $type: group
+  
+  install_packages:
+    $name: Install required packages
+    $type: boolean
+    $default: true
+    $help: Install all required packages to install SAP Netweaver
+  
+  virtual_addresses:
+    $name: Virtual Addresses
+    $type: edit-group
+    $prototype:
+      $key:
+          $type: text
+          $name: IP Address
+      $type: text
+      $name: Hostname
+
+  sidadm_user_checkbox:
+    $name: Create the SAP sidadm user and group before the installation
+    $type: boolean
+    $default: false      
+  sidadm_user:
```

Comment:
Shouldn't `sidadm_user` be optional?
We don't force to have this entry, that for sure.

Reply:
correct, I have a checkbox for it and it is by default `false`. And since this is `group`, I can remove it via pre validation

Reply:
@arbulu89 I have added `optional` to elements inside `sidadm_user`. This will cover the case if sidadm_user is true/false. In case of checkbox true and subelements have empty values error will be generated.
If the user leaves the subelements filled, but does not set the checkbox, the state will be run. It is not possible to do the `pop` element operation at root elements of `netweaver`. So this case is same for other checkbox cases under `netweaver`